crates/wiggle/wasmtime/macro/src/lib.rs

```diff
@@ -209,13 +209,17 @@ fn generate_func(
                             #handle_early_error
                         }
                     };
-                    let mem = #runtime::WasmtimeGuestMemory::new(mem);
-                    #target_module::#name_ident(
-                        &mut my_cx.borrow_mut(),
-                        &mem,
-                        #(#arg_names),*
-                    )
-                }
+            let mem = #runtime::WasmtimeGuestMemory::new(mem);
+            let result = #target_module::#name_ident(
+            &mut my_cx.borrow_mut(),
+            &mem,
+            #(#arg_names),*
+            );
+            match result {
+                Ok(r) => {return Ok(r.into());},
+                Err(err) => { return Err(wasmtime::Trap::new(err)); },
+            }
+        }
             }
```

Comment:
I don't have experience with this combination of procedural macros and `cargo fmt`.  Any advice here?

Reply:
Ah unfortunately `cargo fmt` ignores everything inside of procedural macros (since they're opaque as to what they're supposed to be), so we're left to manually format everything inside.

Reply:
Thanks!  It's looking better now.

Reply:
I appreciate your patience @alexcrichton with all the formatting errors in this PR.  I wasn't aware that `cargo fmt` ignores everything inside of proc macros.  #life-long-learning.